swagger.yaml

```diff
@@ -0,0 +1,81 @@
+swagger: '2.0'
+info:
+  description: 'OSBuild Management'
+  version: 1.0.0
+  title: OSBuildManagement
+  contact:
+    name: Flotta flotta
+    url: https://github.com/project-flotta
+    email: flotta@redhat.com
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+basePath: /api/osbuild/v1
+tags:
+  - name: osbuildconfig
+    description: OSBuildConfig CRD
+
+schemes:
+  - https
+consumes:
+  - application/json
+produces:
+  - application/json
+
+securityDefinitions:
+  agentAuth:
+    type: apiKey
+    in: header
+    name: X-Secret-Key
+
+paths:
+  /osbuildconfig/{osbuildconfig_namespace_name}/{osbuildconfig_cr_name}/webhooks/{secret}/generic:
+    post:
+      description: Triggering webhook builds for OSBuildConfig CRs
+      operationId: OSBuildConfigWebhookTriggers
+      tags:
+        - osbuildconfig
+      parameters:
+        - in: path
+          name: osbuildconfig_namespace_name
+          description: OSBuilfConfig namespace name
+          type: string
+          required: true
+        - in: path
+          name: osbuildconfig_cr_name
+          description: OSBuildConfig name
+          type: string
+          required: true
+        - in: path
+          name: secret
+          description: secret name
```

Comment:
what is the content of the secret? does it contain a sensitive information? should it be specified in the path?

Reply:
Please see: https://docs.openshift.com/container-platform/4.7/cicd/builds/triggering-builds-build-hooks.html#builds-using-generic-webhooks_triggering-builds-build-hooks